mattermost/desktop
# Implement auto updater for Windows and Mac

Before submitting, please confirm you've

Please provide the following information:

**Summary**
Implement auto updater.

This is a first prototyping of auto-updater. At this time, this PR only considers the Windows installer.

NSIS installer is introduced. It integrates 32-bit and 64-bit installers into one installer file. And the new installation folder becomes `%USERPROFILE%\AppData\Local\Programs\mattermost` (Per user install). We can also choose `C:\Program Files` (Per machine install) by modifying electron-builder settings.

<img width="358" alt="updater-dialog" src="https://user-images.githubusercontent.com/1412057/30032314-31557946-91d0-11e7-9a17-be17eb8667f9.png">


**Issue link**
#15 

**Test Cases**
I prepared a version, v9.9.9 for testing: https://circleci.com/gh/yuya-oc/desktop/366#artifacts

A. simple case with CircleCI

1. Install the app.
2. Edit `%USERPROFILE%\AppData\Local\Programs\mattermost\resources\app-update.yml`
    - Set `'https://366-67276967-gh.circle-artifacts.com/0/tmp/circle-artifacts.0utXb6g/'` to the value of `url`
3. The app will check updates to CircleCI URL.

B. manual testing for various cases.
1. Prepare a http server to locate `latest.yml` and `mattermost-setup-9.9.9.exe`.
    Example:  [`http-server`](https://www.npmjs.com/package/http-server)
    1. Install Node.js.
    2. Execute `npm install -g http-server`
    3. Execute `http-server -p 8081` under the folder where artifacts are located.
2. The app will check updates to `http://localhost:8081/latest.yml`.
3. You can change the test version and release date by editing `latest.yml`.

**Additional Notes**
https://circleci.com/gh/yuya-oc/desktop/365#artifacts
